config/backend-config/backend_config_test.go

```diff
@@ -84,8 +95,86 @@ var sampleBackendConfig2 = ConfigT{
 	},
 }
 
+var (
+	hold bool
+	db   *sql.DB
+)
+
+func TestMain(m *testing.M) {
+	flag.BoolVar(&hold, "hold", false, "hold environment clean-up after test execution until Ctrl+C is provided")
+	flag.Parse()
+
+	// hack to make defer work, without being affected by the os.Exit in TestMain
+	os.Exit(run(m))
+}
+
+func run(m *testing.M) int {
```

Comment:
for tests it would be great if we had a way to disable cache, or use a mock cache so that we don't need to have a real database as a prerequisite for every test scenario.